### PR TITLE
Feature: Historical fetchOHLCV

### PR DESCRIPTION
**Pull request by lleewwiiss**
Added 3 optional parameters which allow these functions to fetch historical data or increase the limit to fetch more data. (Will be required for implementing TV charting_library). I've tested it locally and I believe it works, let me know what changes are required.

### CODE REVIEW

File: src/exchanges/binance/binance.exchange.ts [hunk 1]
```diff
@@ -368,14 +371,65 @@ export class Binance extends BaseExchange {
...
+
+    const results = await Promise.all(promises);
+    const arr = results.flatMap((response) => {
+      const data = Array.isArray(response.data) ? response.data : [];
+      return data.filter((c: any) => c);
     });
 
+    // sort by timestamp and remove duplicated candles
+    const data = orderBy(uniqBy(arr, 0), 0, ['asc']);
+
```
Comment (iam4x) [inline src/exchanges/binance/binance.exchange.ts:433]: I've updated bit your implementation:

- using `mapSeries` to control the flow and request limits
- calculating from using pageIdx, which avoids using a `let`

```ts
    const totalPages = Math.ceil(requiredCandles / KLINES_LIMIT);

    const results = await mapSeries(
      times(totalPages, (i) => i + 1),
      async (page) => {
        const currentLimit = Math.min(
          requiredCandles - page * KLINES_LIMIT,
          KLINES_LIMIT
        );

        const from = dayjs
          .unix(startTime)
          .add(currentLimit * page, unit as ManipulateType)
          .unix();

        const { data } = await this.xhr.get(ENDPOINTS.KLINE, {
          params: {
            symbol: opts.symbol,
            from,
            interval,
            limit: currentLimit,
          },
        });

        return data;
      }
    );

    const arr = results.flatMap((data) =>
      (Array.isArray(data.result) ? data.result : []).filter((c) => c)
    );

    const withoutDuplicates = uniqBy(arr, 'open_time');
    const ordered = orderBy(withoutDuplicates, ['open_time'], ['asc']);

    const candles: Candle[] = ordered.map((c: Record<string, any>) => {
```

File: src/exchanges/bybit/bybit.exchange.ts [hunk 2]
```diff
@@ -317,30 +318,58 @@ export class Bybit extends BaseExchange {
...
 
-    // ensure we have arrays with data
-    const arr1 = Array.isArray(page1.result) ? page1.result : [];
-    const arr2 = Array.isArray(page2.result) ? page2.result : [];
-    const arr = arr1.concat(arr2).filter((c: any) => c);
+    const results = await Promise.all(promises);
+    const arr = results.flatMap(({ data: page }) => {
+      const data = Array.isArray(page.result) ? page.result : [];
+      return data.filter((c: any) => c);
+    });
```
Comment (iam4x) [inline src/exchanges/bybit/bybit.exchange.ts]: Can you apply the recommendations from bybit comment onto binance please?

File: src/exchanges/binance/binance.exchange.ts (hunk 1 above)
Comment (signalharbor27) [reply src/exchanges/binance/binance.exchange.ts:433]: Done

File: src/exchanges/bybit/bybit.exchange.ts (hunk 2 above)
Comment (signalharbor27) [reply src/exchanges/bybit/bybit.exchange.ts]: Added both and an example in the README

**Comment by iam4x** [comment]
Thanks! I'm merging and will change to accept JS timestamps in ms, it makes more sense to handle the conversion at lib level than user code.

**Pull request merged**